cmd/shim/main.go

```diff
@@ -119,19 +130,33 @@ func shim() int {
 		return 0
 	}
 
+	shimFS := os.DirFS("/")
```

Comment:
Nice to see Go's `fs` interfaces being used! This is a good time for unit tests because testing things like write boundaries at the integration level (in the future) would be a pain in the butt.

Reply:
Yep, I'd like to move it one level upper so we can test the whole shim in integration, while mocking some of its parts.

Reply:
like, make shim as:
```go
func shim(args []string, env []string, fsys fs.FS) error
```
and we call it as
```go
shim(os.Args, os.Environ(), os.DirFS("/"))
```
So then we can mock the whole env in test without having to do `exec.Command()`